src/test/java/io/debezium/connector/yugabytedb/YugabyteDBStreamConsistencyTest.java

```diff
@@ -180,4 +184,130 @@ public void recordsShouldStreamInConsistentOrderOnly() throws Exception {
+
+            // Inserting the index of the record for department table at its appropriate position.
+            departmentIndices.add((int) totalCount);
+            ++totalCount;
+
+            for (int j = employeeId; j <= employeeId + employeeBatchSize - 1; ++j) {
+                TestHelper.execute(String.format("INSERT INTO employee VALUES (%d, 'emp no %d', %d);", j, j, departmentId));
+                employeeIndices.add((int) totalCount);
+                ++totalCount;
+                for (int k = contractId; k <= contractId + contractBatchSize - 1; ++k) {
+                    TestHelper.execute(String.format("INSERT INTO contract VALUES (%d, 'contract no %d', %d);", k, k, j /* employee fKey */));
+                    contractIndices.add((int) totalCount);
+                    ++totalCount;
+
+                    for (int l = addressId; l <= addressId + addressBatchSize - 1; ++l) {
+                        TestHelper.execute(String.format("INSERT INTO address VALUES (%d, 'address no %d', %d);", l, l, k /* contract fKey */));
+                        addressIndices.add((int) totalCount);
+                        ++totalCount;
+                    }
+                    // Increment addressId for next iteration.
+                    addressId += addressBatchSize;
+                }
+                // Increment contractId for next iteration.
+                contractId += contractBatchSize;
+            }
+
+            // Increment employeeId for the next iteration
+            employeeId += employeeBatchSize;
+
+            // Increment department ID for more iterations
+            ++departmentId;
+        }
+
+        // Dummy wait
+        TestHelper.waitFor(Duration.ofSeconds(25));
+
+        List<SourceRecord> recordsToAssert = new ArrayList<>();
+
+        final long total = totalCount;
+        AtomicLong totalConsumedRecords = new AtomicLong();
+        try {
+            Awaitility.await()
+                    .atMost(Duration.ofSeconds(600))
+                    .until(() -> {
+                        int consumed = super.consumeAvailableRecords(record -> {
+                            LOGGER.debug("The record being consumed is " + record);
+                            Struct s = (Struct) record.value();
+                            recordsToAssert.add(record);
+                        });
+                        if (consumed > 0) {
+                            totalConsumedRecords.addAndGet(consumed);
+                            LOGGER.info("Consumed " + totalConsumedRecords.get() + " records");
+                        }
+
+                        return recordsToAssert.size() == total;
+                    });
+        } catch (ConditionTimeoutException exception) {
+            fail("Failed to consume " + totalCount + " records in 600 seconds, consumed " + totalConsumedRecords.get(), exception);
+        }
+
+        assertEquals(total, recordsToAssert.size());
```

Comment:
Do we just assert the count?

Reply:
No, we assert the record's position as well later in the test.